rust/src/tx_builder.rs

```diff
@@ -583,6 +585,47 @@ impl TransactionBuilder {
         self.collateral = collateral.clone();
     }
 
+    pub fn set_collateral_return(&mut self, collateral_return: &TransactionOutput) {
+        self.collateral_return = Some(collateral_return.clone());
+    }
+
+    /// This function will set the collateral-return value and then auto-calculate and assign
+    /// the total collateral coin value. Will raise an error in case no collateral inputs are set
+    /// or in case the total collateral value will have any assets in it except coin.
+    pub fn set_collateral_return_and_total(&mut self, collateral_return: &TransactionOutput) -> Result<(), JsError> {
+        self.set_collateral_return(collateral_return);
+        let collateral = &self.collateral;
+        if collateral.len() == 0 {
+            return Err(JsError::from_str("Cannot calculate total collateral value when collateral inputs are missing"));
+        }
+        let col_input_value: Value = collateral.total_value()?;
+        let total_col: Value = col_input_value.checked_sub(&collateral_return.amount())?;
+        if total_col.multiasset.is_some() {
+            return Err(JsError::from_str("Total collateral value cannot contain assets!"));
+        }
+        self.total_collateral = Some(total_col.coin);
```

Comment:
Is there restriction for minimal total_collateral value ?

Reply:
No hard restriction apart from it being like at least 1 lovelace prolly, or nothing, but there's a protocol logic of course that it might be equal to a certain percentage  (more than 100) of the transaction fee. Later after the hardfork we will update the fee-calculation logic to calculate both fee and collateral at the same time, but it's more complicated, requires more time, and is lower priority.

Reply:
In the setter the logic is supposed to be the same as with the fee setter - there are protocol requirements for the value and we might provide helpers to calculate it, but we allow you to fill the structure with whatever values you want if you are using manual setters.